providers/os/id/aws/aws.go

```diff
@@ -24,9 +25,24 @@ func readValue(conn shared.Connection, fPath string) string {
 	return string(content)
 }
 
-func Detect(conn shared.Connection, p *inventory.Platform, smbiosMgr smbios.SmBiosManager) (string, string, []string) {
+func Detect(conn shared.Connection, p *inventory.Platform) (string, string, []string) {
 	var values []string
-	if p.IsFamily(inventory.FAMILY_LINUX) {
+	if conn.Type() == shared.Type_FileSystem {
```

Comment:
you should double check if thats the type we give to the device connection too

Reply:
At first, I thought the type was `shared.Type_Device` but after lots of debugging (difficult because you have to test on a "runner") I figured out this is the type we land on.

I think it is because we nest two types inside the device connection struct https://github.com/mondoohq/cnquery/blob/main/providers/os/connection/device/device_connection.go#L34-L35 
 and we have config and asset there and in the nested fs struct https://github.com/mondoohq/cnquery/blob/main/providers/os/connection/fs/filesystem.go#L55-L58

Somehow we endup having the nested struct instead of the main device connection struct.